app/services/consent_service.rb

```diff
@@ -15,12 +15,27 @@ def process_consent_change(sms_message)
     end
   end
 
+  def check_consent(recipient, program)
+    default_consent = true
+    latest_consent_change = ConsentChange
+      .joins(:sms_message)
```

Comment:
Minor: this may be better as a scope in `ConsentChange` model?

Reply:
See https://github.com/codeforamerica/notifications/blob/main/spec/factories/recipient.rb#L3

Reply:
I moved this into a ConsentChange class method per the comment [here](https://guides.rubyonrails.org/active_record_querying.html#passing-in-arguments). Let me know if you think a scope with an argument would be more idiomatic!

Reply:
Not too particular, but I do think a scope would be more idiomatic. Also,

> A scope will always return an ActiveRecord::Relation object, even if the conditional evaluates to false, whereas a class method, will return nil. This can cause NoMethodError when chaining class methods with conditionals, if any of the conditionals return false.

Reply:
Took another crack at it. Moving the whole method directly into a scope doesn't work since scopes return relations rather than single objects, so at best I'd have to put .last on every call site which I don't want to do. Instead I took a queue from [this](https://andycroll.com/ruby/a-scope-should-return-a-scope/) and moved the filtering logic into a scope and the ordering & selection logic into a class method.